### ISSUE TITLE
Switch to Pikaur instead of Yay

### ISSUE THREAD

**Issue by CummingCowGirl** [topgrade-rs/topgrade#992]
I just ran Topgrade and got the below and would like to know if I can make Topgrade use Pikaur instead of Yay? I prefer Pikaur for my AUR helper anyway. Thanks

~~~I just ran Topgrade to do some updates that garuda-update doesn't do and Sonarr and Radarr had updates and after building the files I got the message "Yay exited with error code 1". I have no idea what is up with that cause Yay has been performing perfectly. I thought it might be a issue with the particular updates so I ran pikaur -S radarr-develop sonarr-develop and no issues at all.

Garuda Linux completely up to date Yay version 12.4.2-1 Topgrad version 16.0.1-1~~~

**Comment by Jisu-Woniu** [comment]
The config has already covered it:

https://github.com/topgrade-rs/topgrade/blob/aebc035ec06fcb1ea476d78f8bd465385df636dd/config.example.toml#L145-L149

**Comment by CummingCowGirl** [comment]
Sorry for the late reply but topgrade stated behaving where the AUR is concerned til today. Keeps wanting to use paru when I though yay was the default if installed. It errored out trying to update prowlarr-nightly-bin. Updating prowlarr using pikaur -S I had no issue.  I was wondering if there was any progress on this? Thanks

**Comment by CummingCowGirl** [comment]
`~  topgrade                                                                                       ✔  04:54:48 AM

── 04:54:51 - System update ────────────────────────────────────────────────────
:: Synchronizing package databases...
garuda is up to date
herecura is up to date
core                                      116.5 KiB   294 KiB/s 00:00 [---------------------------------------] 100%
extra is up to date
multilib is up to date
chaotic-aur is up to date
cachyos-znver4 is up to date
cachyos-core-znver4 is up to date
cachyos-extra-znver4 is up to date
cachyos is up to date
arcolinux_repo                             49.3 KiB   331 KiB/s 00:00 [---------------------------------------] 100%
arcolinux_repo_3party                     145.3 KiB   850 KiB/s 00:00 [---------------------------------------] 100%
arcolinux_repo_xlarge                      19.8 KiB   133 KiB/s 00:00 [---------------------------------------] 100%
nemesis_repo is up to date
endeavouros is up to date
Reborn-OS is up to date
spawn pacman -Su
:: Starting full system upgrade...
warning: pacman: local (7.0.0.r6.gc685ae6-2) is newer than core (7.0.0.r6.gc685ae6-1)
there is nothing to do

--> Updating AUR packages with paru..
:: Looking for devel upgrades...
:: Resolving dependencies...
:: Calculating conflicts...
:: Calculating inner conflicts...

Repo (4)                          Old Version    New Version      Make Only
extra/dotnet-sdk-6.0                             6.0.36.sdk136-2  Yes
extra/dotnet-targeting-pack-6.0                  6.0.36.sdk136-2  Yes
extra/netstandard-targeting-pack                 9.0.1.sdk102-1   Yes
extra/yarn                                       1.22.22-2        Yes

Aur (1)                           Old Version    New Version      Make Only
aur/radarr-develop                5.18.3.9669-1  5.18.4.9674-1    No

:: Proceed to review? [Y/n]: n

Paru exited with error code 1


System updated! 🐧`

**Comment by Jisu-Woniu** [comment]
Don't understand what was wrong. Could you please paste your `topgrade.toml` config and run `topgrade --verbose` instead.

I think maybe topgrade invoked `garuda-update` which ran `paru` to update AUR packages.

**Comment by CummingCowGirl** [comment]
~~~
# Include any additional configuration file(s)
# [include] sections are processed in the order you write them
# Files in $CONFIG_DIR/topgrade.d/ are automatically included before this file
[include]
# paths = ["/etc/topgrade.toml"]


[misc]
# Run `sudo -v` to cache credentials at the start of the run
# This avoids a blocking password prompt in the middle of an unattended run
# (default: false)
# pre_sudo = false

# Sudo command to be used
# sudo_command = "sudo"

# Disable specific steps - same options as the command line flag
# disable = ["system", "emacs"]

# Ignore failures for these steps
# ignore_failures = ["powershell"]

# List of remote machines with Topgrade installed on them
# remote_topgrades = ["toothless", "pi", "parnas"]

# Path to Topgrade executable on remote machines
# remote_topgrade_path = ".cargo/bin/topgrade"

# Arguments to pass to SSH when upgrading remote systems
# ssh_arguments = "-o ConnectTimeout=2"

# Arguments to pass tmux when pulling Repositories
# tmux_arguments = "-S /var/tmux.sock"

# Do not set the terminal title (default: true)
# set_title = true

# Display the time in step titles (default: true)
# display_time = true

# Don't ask for confirmations (no default value)
# assume_yes = true

# Do not ask to retry failed steps (default: false)
# no_retry = true

# Run inside tmux (default: false)
# run_in_tmux = true

# Changes the way topgrade interacts with
# the tmux session, creating the session
# and only attaching to it if not inside tmux
# (default: "attach_if_not_in_session", allowed values: "attach_if_not_in_session", "attach_always")
# tmux_session_mode = "attach_if_not_in_session"

# Cleanup temporary or old files (default: false)
# cleanup = true

# Send a notification for every step (default: false)
# notify_each_step = false

# Skip sending a notification at the end of a run (default: false)
# skip_notify = true

# The Bash-it branch to update (default: "stable")
# bashit_branch = "stable"

# Run specific steps - same options as the command line flag
# only = ["system", "emacs"]

# Whether to self update
#
# this will be ignored if the binary is built without self update support
#
# available also via setting the environment variable TOPGRADE_NO_SELF_UPGRADE)
# no_self_update = true

# Extra tracing filter directives
# These are prepended to the `--log-filter` argument
# See: https://docs.rs/tracing-subscriber/latest/tracing_subscriber/filter/struct.EnvFilter.html#directives
# log_filters = ["topgrade::command=debug", "warn"]


# Commands to run before anything
[pre_commands]
# "Emacs Snapshot" = "rm -rf ~/.emacs.d/elpa.bak && cp -rl ~/.emacs.d/elpa ~/.emacs.d/elpa.bak"


# Commands to run after anything
[post_commands]
# "Emacs Snapshot" = "rm -rf ~/.emacs.d/elpa.bak && cp -rl ~/.emacs.d/elpa ~/.emacs.d/elpa.bak"


# Custom commands
[commands]
# "Python Environment" = "~/dev/.env/bin/pip install -i https://pypi.python.org/simple -U --upgrade-strategy eager jupyter"
# "Custom command using interactive shell (unix)" = "-i vim_upgrade"


[python]
# enable_pip_review = true                         ###disabled by default
# enable_pip_review_local = true                   ###disabled by default
# enable_pipupgrade = true                         ###disabled by default
# pipupgrade_arguments = "-y -u --pip-path pip"    ###disabled by default

# For the poetry step, by default, Topgrade skips its update if poetry is not 
# installed with the official script. This configuration entry forces Topgrade 
# to run the update in this case.
#
# (default: false)
# poetry_force_self_update = true


[composer]
# self_update = true


[brew]
# For the BrewCask step
# If `Repo Cask Upgrade` exists, then use the `-a` option.
# Otherwise, use the `--greedy` option.
# greedy_cask = true

# For the BrewCask step
# If `Repo Cask Upgrade` does not exist, then use the `--greedy_latest` option.
# NOTE: the above entry `greedy_cask` contains this entry, though you can enable
# both of them, they won't clash with each other.
# greedy_latest = true

# For the BrewCask step
# If `Repo Cask Upgrade` does not exist, then use the `--greedy_auto_updates` option.
# NOTE: the above entry `greedy_cask` contains this entry, though you can enable
# both of them, they won't clash with each other.
# greedy_auto_updates = true

# For the BrewFormula step
# Execute `brew autoremove` after the step.
# autoremove = true

# For the BrewFormula step
# Upgrade formulae built from the HEAD branch; `brew upgrade --fetch-HEAD`
# fetch_head = true


[linux]
# Arch Package Manager to use.
# Allowed values:
#   autodetect, aura, garuda_update, pacman, pamac, paru, pikaur, trizen, yay
# arch_package_manager = "pacman"

# Arguments to pass yay (or paru) when updating packages
# yay_arguments = "--nodevel"

# Arguments to pass dnf when updating packages
# dnf_arguments = "--refresh"

# aura_aur_arguments = "-kx"

# aura_pacman_arguments = ""
# garuda_update_arguments = ""

# show_arch_news = true

# trizen_arguments = "--devel"

# pikaur_arguments = ""

# pamac_arguments = "--no-devel"

# enable_tlmgr = true

# emerge_sync_flags = "-q"

# emerge_update_flags = "-uDNa --with-bdeps=y world"

# redhat_distro_sync = false

# suse_dup = false

# rpm_ostree = false

# For Fedora/CentOS/RHEL Atomic variants, if `bootc` is available and this configuration entry is set to true, use
# it to do the update - Will also supercede rpm-ostree if enabled
# (default: false)
# bootc = false

# nix_arguments = "--flake"

# nix_env_arguments = "--prebuilt-only"

# Extra Home Manager arguments
# home_manager_arguments = ["--flake", "file"]


[git]
# How many repos to pull at max in parallel
# max_concurrency = 5

# Additional git repositories to pull
# repos = [
#     "~/src/*/",
#     "~/.config/something"
# ]

# Don't pull the predefined git repos
# pull_predefined = false

# Arguments to pass Git when pulling Repositories
# arguments = "--rebase --autostash"


[windows]
# Manually select Windows updates
# accept_all_updates = false

# open_remotes_in_new_terminal = true

# wsl_update_pre_release = true

# wsl_update_use_web_download = true

# Causes Topgrade to rename itself during the run to allow package managers
# to upgrade it. Use this only if you installed Topgrade by using a package
# manager such as Scoop or Cargo
# self_rename = true


[npm]
# Use sudo if the NPM directory isn't owned by the current user
# use_sudo = true


[yarn]
# Run `yarn global upgrade` with `sudo`
# use_sudo = true


[deno]
# Upgrade deno executable to the given version.
# version = "stable"


[vim]
# For `vim-plug`, execute `PlugUpdate!` instead of `PlugUpdate`
# force_plug_update = true


[firmware]
# Offer to update firmware; if false just check for and display available updates
# upgrade = true


[vagrant]
# Vagrant directories
# directories = []

# power on vagrant boxes if needed
# power_on = true

# Always suspend vagrant boxes instead of powering off
# always_suspend = true


[flatpak]
# Use sudo for updating the system-wide installation
# use_sudo = true


[distrobox]
# use_root = false

# containers = ["archlinux-latest"]
[containers]
# Specify the containers to ignore while updating (Wildcard supported)
# ignored_containers = ["ghcr.io/rancher-sandbox/rancher-desktop/rdx-proxy:latest", "docker.io*"]
# Specify the runtime to use for containers (default: "docker", allowed values: "docker", "podman")
# runtime = "podman"

[lensfun]
# If disabled, Topgrade invokes `lensfun‑update‑data` without root priviledge,
# then the update will be only available to you. Otherwise, `sudo` is required,
# and the update will be installed system-wide, i.e., available to all users.
# (default: false)
# use_sudo = false

[julia]
# If disabled, Topgrade invokes julia with the --startup-file=no CLI option.
#
# This may be desirable to avoid loading outdated packages with "using" directives
# in the startup file, which might cause the update run to fail.
# (default: true)
# startup_file = true
~~~

~~~
~  topgrade --verbose                                                                             ✔  06:12:17 AM
DEBUG Current system locale is en-US
DEBUG Configuration at /home/cccp/.config/topgrade.toml
DEBUG Loaded configuration: ConfigFile { include: Some(Include { paths: None }), misc: Some(Misc { pre_sudo: None, sudo_command: None, disable: None, ignore_failures: None, remote_topgrades: None, remote_topgrade_path: None, ssh_arguments: None, tmux_arguments: None, set_title: None, display_time: None, assume_yes: None, no_retry: None, run_in_tmux:None, tmux_session_mode: None, cleanup: None, notify_each_step: None, skip_notify: None, bashit_branch: None, only: None, no_self_update: None, log_filters: None }), pre_commands: Some({}), post_commands: Some({}), commands: Some({}), python: Some(Python { enable_pip_review: None, enable_pip_review_local: None, enable_pipupgrade: None, pipupgrade_arguments: None, poetry_force_self_update: None }), composer: Some(Composer { self_update: None }), brew: Some(Brew { greedy_cask: None, greedy_latest: None, greedy_auto_updates: None, autoremove: None, fetch_head: None }), linux: Some(Linux { yay_arguments: None, aura_aur_arguments: None, aura_pacman_arguments: None, arch_package_manager: None, show_arch_news: None, garuda_update_arguments: None, trizen_arguments: None, pikaur_arguments: None, pamac_arguments: None, dnf_arguments: None, nix_arguments: None, nix_env_arguments: None, apt_arguments: None, enable_tlmgr: None, redhat_distro_sync: None, suse_dup: None, rpm_ostree: None, bootc: None, emerge_sync_flags: None, emerge_update_flags: None, home_manager_arguments: None }), git: Some(Git { max_concurrency: None, arguments: None, repos: None, pull_predefined: None }), containers: Some(Containers { ignored_containers: None, runtime: None }), windows: Some(Windows { accept_all_updates: None, self_rename: None, open_remotes_in_new_terminal: None, wsl_update_pre_release: None, wsl_update_use_web_download: None }), npm: Some(NPM { use_sudo: None }), yarn: Some(Yarn { use_sudo: None }), deno: Some(Deno { version: None }), vim: Some(Vim { force_plug_update: None }), firmware: Some(Firmware { upgrade: None }), vagrant: Some(Vagrant { directories: None, power_on: None, always_suspend: None }), flatpak: Some(Flatpak { use_sudo: None }), distrobox: Some(Distrobox { use_root: None, containers: None }), lensfun: Some(Lensfun { use_sudo: None }), julia: Some(JuliaConfig { startup_file: None }) }
DEBUG Version: 16.0.2
DEBUG OS: x86_64-unknown-linux-gnu
DEBUG Args { inner: ["topgrade", "--verbose"] }
DEBUG Binary path: Ok("/usr/bin/topgrade")
DEBUG self-update Feature Enabled: false
DEBUG Configuration: Config { opt: CommandLineArgs { edit_config: false, show_config_reference: false, run_in_tmux: false, cleanup: false, dry_run: false, no_retry: false, disable: [], only: [], custom_commands: [], env: [], verbose: true, keep_at_end: false, skip_notify: false, yes: None, disable_predefined_git_repos: false, config: None, remote_host_limit: None, show_skipped: false, log_filter: "warn", gen_completion: None, gen_manpage: false, no_self_update: false }, config_file: ConfigFile { include: Some(Include { paths: None }), misc: Some(Misc { pre_sudo: None, sudo_command: None, disable: None, ignore_failures: None, remote_topgrades: None, remote_topgrade_path: None, ssh_arguments: None, tmux_arguments: None, set_title: None, display_time: None, assume_yes: None, no_retry: None, run_in_tmux: None, tmux_session_mode: None, cleanup: None, notify_each_step: None, skip_notify: None, bashit_branch: None, only: None, no_self_update: None, log_filters: None }), pre_commands: Some({}), post_commands: Some({}), commands: Some({}), python: Some(Python { enable_pip_review: None, enable_pip_review_local: None, enable_pipupgrade: None, pipupgrade_arguments: None, poetry_force_self_update: None }), composer: Some(Composer { self_update: None }), brew: Some(Brew { greedy_cask: None, greedy_latest: None, greedy_auto_updates: None, autoremove: None, fetch_head: None }), linux: Some(Linux {yay_arguments: None, aura_aur_arguments: None, aura_pacman_arguments: None, arch_package_manager: None, show_arch_news: None, garuda_update_arguments: None, trizen_arguments: None, pikaur_arguments: None, pamac_arguments: None, dnf_arguments: None, nix_arguments: None, nix_env_arguments: None, apt_arguments: None, enable_tlmgr: None, redhat_distro_sync: None, suse_dup: None, rpm_ostree: None, bootc: None, emerge_sync_flags: None, emerge_update_flags: None, home_manager_arguments: None }), git: Some(Git { max_concurrency: None, arguments: None, repos: None, pull_predefined: None}), containers: Some(Containers { ignored_containers: None, runtime: None }), windows: Some(Windows { accept_all_updates: None, self_rename: None, open_remotes_in_new_terminal: None, wsl_update_pre_release: None, wsl_update_use_web_download: None }), npm: Some(NPM { use_sudo: None }), yarn: Some(Yarn { use_sudo: None }), deno: Some(Deno { version: None }), vim: Some(Vim { force_plug_update: None }), firmware: Some(Firmware { upgrade: None }), vagrant: Some(Vagrant { directories: None, power_on: None, always_suspend: None }), flatpak: Some(Flatpak { use_sudo: None }), distrobox:Some(Distrobox { use_root: None, containers: None }), lensfun: Some(Lensfun { use_sudo: None }), julia: Some(JuliaConfig { startup_file: None }) }, allowed_steps: [AM, AppMan, Asdf, Atom, Aqua, Audit, AutoCpufreq, Bin, Bob, BrewCask,BrewFormula, Bun, BunPackages, Cargo, Certbot, Chezmoi, Chocolatey, Choosenim, ClamAvDb, Composer, Conda, ConfigUpdate, Containers, CustomCommands, DebGet, Deno, Distrobox, DkpPacman, Dotnet, Elan, Emacs, Firmware, Flatpak, Flutter, Fossil, Gcloud, Gem, Ghcup, GithubCliExtensions, GitRepos, GnomeShellExtensions, Go, Guix, Haxelib, Helm, HomeManager, Jetpack, Julia, Juliaup, Kakoune, Helix, Krew, Lure, Lensfun, Macports, Mamba, Miktex, Mas, Maza, Micro, MicrosoftStore, Mise, Myrepos, Nix, Node, Opam, Pacdef, Pacstall, Pearl, Pip3, PipReview, PipReviewLocal, Pipupgrade, Pipx, Pixi, Pkg, Pkgin, PlatformioCore, Pnpm, Poetry, Powershell, Protonup, Pyenv, Raco, Rcm, Remotes, Restarts, Rtcl, RubyGems, Rustup, Rye, Scoop, Sdkman, SelfUpdate, Sheldon, Shell, Snap, Sparkle, Spicetify, Stack, Stew, System, Tldr, Tlmgr, Tmux, Toolbx, Uv, Vagrant, Vcpkg, Vim, VoltaPackages, Vscode, Waydroid, Winget, Wsl, WslUpdate, Xcodes, Yadm, Yarn,Zvm] }
DEBUG Cannot find "pwsh"
DEBUG Cannot find "powershell"
DEBUG Path "/home/cccp/.config/emacs" doesn't exist
DEBUG Path "/home/cccp/.emacs.d" doesn't exist
DEBUG Cannot find "doas"
DEBUG Detected "/usr/bin/sudo" as "sudo"
DEBUG Step "packer.nu"
DEBUG Step "System update"

── 06:12:30 - System update ────────────────────────────────────────────────────
DEBUG Cannot find "powerpill"
DEBUG Detected "/usr/bin/garuda-update" as "garuda-update"
DEBUG Executing command `/usr/bin/garuda-update`
[sudo] password for cccp:
:: Synchronizing package databases...
garuda                                     52.6 KiB  88.6 KiB/s 00:01 [---------------------------------------] 100%
herecura                                   11.7 KiB  4.76 KiB/s 00:02 [---------------------------------------] 100%
core is up to date
extra                                       7.5 MiB  6.69 MiB/s 00:01 [---------------------------------------] 100%
multilib is up to date
chaotic-aur                               650.8 KiB   985 KiB/s 00:01 [---------------------------------------] 100%
cachyos-znver4                            107.5 KiB  72.0 KiB/s 00:01 [---------------------------------------] 100%
cachyos-core-znver4 is up to date
cachyos-extra-znver4                        3.9 MiB  1781 KiB/s 00:02 [---------------------------------------] 100%
cachyos                                   481.8 KiB   286 KiB/s 00:02 [---------------------------------------] 100%
arcolinux_repo                             49.3 KiB   155 KiB/s 00:00 [---------------------------------------] 100%
arcolinux_repo_3party                     145.1 KiB   505 KiB/s 00:00 [---------------------------------------] 100%
arcolinux_repo_xlarge                      20.5 KiB  97.4 KiB/s 00:00 [---------------------------------------] 100%
nemesis_repo is up to date
endeavouros is up to date
Reborn-OS is up to date
proaudio is up to date
spawn pacman -Su
:: Starting full system upgrade...
warning: pacman: local (7.0.0.r6.gc685ae6-2) is newer than core (7.0.0.r6.gc685ae6-1)
resolving dependencies...
looking for conflicting packages...

Package (2)          Old Version  New Version  Net Change  Download Size

extra/python-pillow  11.0.0-3     11.1.0-1       0.01 MiB       0.84 MiB
extra/ugrep          7.2.1-1      7.2.2-1        0.01 MiB       0.51 MiB

Total Download Size:   1.35 MiB
Total Installed Size:  6.97 MiB
Net Upgrade Size:      0.02 MiB

:: Proceed with installation? [Y/n]
:: Retrieving packages...
ugrep-7.2.2-1-x86_64                      522.2 KiB   740 KiB/s 00:01 [---------------------------------------] 100%
python-pillow-11.1.0-1-x86_64             863.6 KiB  1095 KiB/s 00:01 [---------------------------------------] 100%
Total (2/2)                              1385.7 KiB  1569 KiB/s 00:01 [---------------------------------------] 100%
(2/2) checking keys in keyring                                         [---------------------------------------] 100%
(2/2) checking package integrity                                       [---------------------------------------] 100%
(2/2) loading package files                                            [---------------------------------------] 100%
(2/2) checking for file conflicts                                      [---------------------------------------] 100%
(2/2) checking available disk space                                    [---------------------------------------] 100%
:: Running pre-transaction hooks...
(1/2) Rejecting pacman transaction if running in snapshot...
(2/2) Performing snapper pre snapshots for the following configurations...
==> root: 390
:: Processing package changes...
(1/2) upgrading python-pillow                                          [---------------------------------------] 100%
(2/2) upgrading ugrep                                                  [---------------------------------------] 100%
:: Running post-transaction hooks...
(1/6) Arming ConditionNeedsUpdate...
(2/6) Foreign/AUR package notification
angrysearch 1.0.4-3
audacity-plugins 20171216-1
aur-check-updates-bin 1.1.1-1
bluesky 1.0.1-1
dxreminders 1.21.3-1
flaresolverr-bin 3.3.21-3
google-chrome-dev 134.0.6988.2-1
libelectron 2025.1-1
microsoft-edge-dev-bin 134.0.3096.1-1
nuclear-player-bin 0.6.42-3
opera-beta 117.0.5408.4-1
prowlarr-nightly-bin 1.31.1.4950-1
python-vulkan 1.1.99.1-3
qt5-avif-image-plugin 0.8.1-1
qt6-avif-image-plugin 0.8.1-1
radarr-develop 5.18.4.9674-1
renamemytvseries-qt-bin 2.1.8-1
seatools 5.1.245-1
sharik-bin 3.1-4
sonarr-develop 4.0.12.2900-1
upscaler 1.4.0-2
upscayl-ncnn 20240601.103425-2
webcatalog-bin 62.2.0-1
(3/6) Orphaned package notification...
botan2 2.19.5-2
electron32 32.3.0-1
gcab 1.6-2
http-parser 2.9.4-2
knotifyconfig5 5.116.0-1
libcec 6.0.2-6
libomxil-bellagio 0.9.3-5
libretro-beetle-pce 1243-1
libretro-beetle-pce-fast 1266-1
libretro-beetle-psx 2751-1
libretro-beetle-psx-hw 2751-1
libretro-beetle-supergrafx 1019-1
libretro-blastem 2052-3
libretro-bsnes 1:3180-1
libretro-bsnes-hd 55-1
libretro-bsnes2014 1:577-3
libretro-desmume 6361-1
libretro-dolphin 33044-2
libretro-flycast 6590-1
libretro-gambatte 1064-1
libretro-genesis-plus-gx 2208-1
libretro-kronos 7016-2
libretro-mame 93008-1
libretro-melonds 2016-1
libretro-mesen 2904-1
libretro-mesen-s 916-4
libretro-mgba 9298-2
libretro-mupen64plus-next 1:454-1
libretro-nestopia 1:136-1
libretro-parallel-n64 5259-1
libretro-picodrive 2635-1
libretro-play 8417-1
libretro-ppsspp 42066-1
libretro-retrodream 1104-2
libretro-sameboy 1720-3
libretro-scummvm 158600-1
libretro-snes9x 1:2651-1
libretro-stella2014-git 314.3cc89f0-1
libretro-yabause 3344-1
libsmbios 2.4.3-8
python-geoip 1.3.2-17
python-ordered-set 4.1.0-7
python-pyrate-limiter 3.7.0-2
python-rencode 1.0.6-10
python-setproctitle 1.3.4-1
python-simplejson 3.19.3-1
python-tomli 2.0.1-5
python-toolz 1.0.0-2
python-validate-pyproject 0.23-1
qtkeychain-qt5 0.15.0-1
retroarch-assets-glui 1:522-1
tinyxml 2.6.2-12
waylandpp 1.0.0-3
wpebackend-fdo 1.14.3-1
yajl 2.1.0-6
(4/6) Checking for .pacnew and .pacsave files...
(5/6) Performing snapper post snapshots for the following configurations...
==> root: 391
(6/6) Syncing all file systems...

--> Updating AUR packages with paru..
:: Looking for devel upgrades...
there is nothing to do

System updated! 🐧

DEBUG Step "config-update"
DEBUG Detected "/usr/bin/pacdiff" as "pacdiff"
DEBUG Detected "/usr/bin/vim" as "vim"

── 06:12:59 - Configuration update ─────────────────────────────────────────────
DEBUG Executing command `/usr/bin/sudo '--preserve-env=DIFFPROG' /usr/bin/pacdiff`
DEBUG Step "am"
DEBUG Step "appman"
DEBUG Step "deb-get"
DEBUG Step "toolbx"
DEBUG Step "snap"
DEBUG Detected "/usr/bin/snap" as "snap"

── 06:12:59 - snap ─────────────────────────────────────────────────────────────
DEBUG Executing command `/usr/bin/sudo /usr/bin/snap refresh`
All snaps up to date.
WARNING: There is 1 new warning. See 'snap warnings'.
DEBUG Step "pacstall"
DEBUG Step "pacdef"
DEBUG Step "protonup"
DEBUG Step "distrobox"
DEBUG Step "dkp-pacman"
DEBUG Step "pihole"
DEBUG Step "Firmware upgrades"
DEBUG Detected "/usr/bin/fwupdmgr" as "fwupdmgr"
DEBUG Executing command `uname -r`
DEBUG Uname output: 6.13.1-zen1-1-zen


── 06:12:59 - Firmware upgrades ────────────────────────────────────────────────
DEBUG Executing command `/usr/bin/fwupdmgr refresh`
Updating lvfs
Downloading…             [************************************** ]
Successfully downloaded new metadata: Updates have been published for 0 of 3 local devices
DEBUG Executing command `/usr/bin/fwupdmgr get-updates`
Devices with no available firmware updates:
• CT1000P3SSD8
• CT1000T500SSD8
• SSD 990 EVO 2TB
No updatable devices
DEBUG Step "Restarts"
DEBUG Step "Flatpak"
DEBUG Detected "/usr/bin/flatpak" as "flatpak"

── 06:13:02 - Flatpak User Packages ────────────────────────────────────────────
DEBUG Executing command `/usr/bin/flatpak update --user`
Looking for updates…

Nothing to do.

── 06:13:02 - Flatpak System Packages ──────────────────────────────────────────
DEBUG Executing command `/usr/bin/flatpak update --system`
Looking for updates…

Nothing to do.
DEBUG Step "Brew"
DEBUG Step "LURE"
DEBUG Step "Waydroid"
DEBUG Step "auto-cpufreq"
DEBUG Step "yadm"
DEBUG Step "nix"
DEBUG Step "nix upgrade-nix"
DEBUG Step "guix"
DEBUG Step "home-manager"
DEBUG Step "asdf"
DEBUG Step "mise"
DEBUG Step "pkgin"
DEBUG Step "bun-packages"
DEBUG Step "zr"
DEBUG Detected "/usr/local/bin/zsh" as "zsh"
DEBUG Step "antibody"
DEBUG Detected "/usr/local/bin/zsh" as "zsh"
DEBUG Step "antidote"
DEBUG Detected "/usr/local/bin/zsh" as "zsh"
DEBUG Step "antigen"
DEBUG Detected "/usr/local/bin/zsh" as "zsh"
DEBUG Path "/home/cccp/.zshrc" exists
DEBUG Step "zgenom"
DEBUG Detected "/usr/local/bin/zsh" as "zsh"
DEBUG Path "/home/cccp/.zshrc" exists
DEBUG Step "zplug"
DEBUG Detected "/usr/local/bin/zsh" as "zsh"
DEBUG Path "/home/cccp/.zshrc" exists
DEBUG Step "zinit"
DEBUG Detected "/usr/local/bin/zsh" as "zsh"
DEBUG Path "/home/cccp/.zshrc" exists
DEBUG Step "zi"
DEBUG Detected "/usr/local/bin/zsh" as "zsh"
DEBUG Path "/home/cccp/.zshrc" exists
DEBUG Step "zim"
DEBUG Detected "/usr/local/bin/zsh" as "zsh"
DEBUG Executing command `zsh -c '[[ -n ${ZIM_HOME} ]] && print -n ${ZIM_HOME}'`
DEBUG Command failed: Err(
0: Command failed: `zsh -c '[[ -n ${ZIM_HOME} ]] && print -n ${ZIM_HOME}'`
1: `zsh` failed: exit status: 1 with

Location:
src/steps/zsh.rs:153)
DEBUG Step "oh-my-zsh"
DEBUG Detected "/usr/local/bin/zsh" as "zsh"
DEBUG Step "oh-my-bash"
DEBUG Detected "/usr/bin/bash" as "bash"
DEBUG Step "fisher"
DEBUG Detected "/usr/bin/fish" as "fish"
DEBUG Executing command `/usr/bin/fish -c 'type -t fisher'`
DEBUG Command failed: Err(
0: Command failed: `/usr/bin/fish -c 'type -t fisher'`

Stderr:
type: Could not find 'fisher'
1: `/usr/bin/fish` failed: exit status: 1 with type: Could not find 'fisher'
1:

Location:
src/steps/os/unix.rs:105)
DEBUG Step "bash-it"
DEBUG Step "oh-my-fish"
DEBUG Detected "/usr/bin/fish" as "fish"
DEBUG Step "fish-plug"
DEBUG Detected "/usr/bin/fish" as "fish"
DEBUG Step "fundle"
DEBUG Detected "/usr/bin/fish" as "fish"
DEBUG Step "tmux"
DEBUG Step "TLDR"
DEBUG Detected "/usr/bin/tldr" as "tldr"

── 06:13:02 - TLDR ─────────────────────────────────────────────────────────────
DEBUG Executing command `/usr/bin/tldr --update`
~~~

**Comment by Jisu-Woniu** [comment]
That's it.

```plaintext
── 06:12:30 - System update ────────────────────────────────────────────────────
DEBUG Cannot find "powerpill"
DEBUG Detected "/usr/bin/garuda-update" as "garuda-update"
DEBUG Executing command `/usr/bin/garuda-update`
[sudo] password for cccp:
```

`topgrade` runs `garuda-update`, which invokes `paru`. You should change this config:

```toml
[linux]
# Arch Package Manager to use.
# Allowed values:
#   autodetect, aura, garuda_update, pacman, pamac, paru, pikaur, trizen, yay
# arch_package_manager = "pacman"
```

into

```toml
[linux]
# Arch Package Manager to use.
# Allowed values:
#   autodetect, aura, garuda_update, pacman, pamac, paru, pikaur, trizen, yay
arch_package_manager = "pikaur"
```

**Comment by CummingCowGirl** [comment]
OK thanks, but one question I learned in the past that if I don't run garuda-update first and then topgrade more times than not I would get a partial update warning in the systray from Garuda, so besides editing that part of the.toml can I just remove the garuda_update part? Thanks again.

**Comment by Jisu-Woniu** [comment]
Maybe you should ask that on Garuda forum since there's no such thing of "partial upgrade warning" on normal Arch Linux installs.

As an Arch user, I would say: if you finished all upgrade steps with paru or pikaur successfully, then your system is not in partial upgrade state.

However, you can run garuda-update again to check it.

**Comment by SteveLauC** [comment]
@Jisu-Woniu, appreciate the help!

**Comment by CummingCowGirl** [comment]
Yea I don't believe it's because there is actually a partial update. I believe it pops up because one or more custom fixes in Garuda isn't grabbed by Topgrade, and to be honest I would expect Topgrade to catch them all. To be honest sometimes I think it's just the particular AUR package. Awhile back I ran into this issue, but it only happened with Sonarr, Radarr, Prowlarr. I only noticed cause I install Aur-Check-Updates and run it before I do any updates, the only time then Topgrade would error out was when one or more of those three had a update. No biggie cause I've gotten in the habit of doing the AUR updates first with Pikaur, then Topgrade, then if I need it Garuda-Update.

**Comment by GideonBear** [comment]
Closing this, since it seems to be solved/answered? Feel free to re-open if I'm wrong.
